cmd/spiffe-aws-assume-role/cli/cli_test.go

```diff
@@ -17,39 +20,62 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestParsesSessionDuration(t *testing.T) {
+var requiredArgs = []string{
+	"credentials",
+	"--audience=foo",
+	"--role-arn=bar",
+	"--spiffe-id=baz",
+}
+
+func TestErrorLogging(t *testing.T) {
+	const logFileName = "spiffe-aws-assume-role.log"
+
+	deleteFileIfExists(t, logFileName)
+	defer deleteFileIfExists(t, logFileName)
+
+	badSpiffeId := uuid.New().String()
+
 	args := []string{
 		"credentials",
-		"--session-duration=PT5M",
-		// We only specify the following fields because they're required
 		"--audience=foo",
 		"--role-arn=bar",
-		"--spiffe-id=baz",
+		fmt.Sprintf("--spiffe-id=%s", badSpiffeId),
+		fmt.Sprintf("--log-file-path=%s", logFileName),
 	}
-	context, err := parse(args)
+	err := RunWithDefaultContext(args)
+	require.Error(t, err)
+
+	bytes, err := ioutil.ReadFile(logFileName)
 	require.NoError(t, err)
+	logs := string(bytes)
+	require.True(t, strings.Contains(logs, badSpiffeId))
+}
 
-	cli := context.Model.Target.Interface().(CLI)
-	sessionDuration := cli.Credentials.SessionDuration
-	require.EqualValues(t, 5, sessionDuration.Minutes())
+func TestParsesSessionDuration(t *testing.T) {
+	command := parseTest(t, "--session-duration=PT5M")
+	require.EqualValues(t, 5, command.SessionDuration.Minutes())
 }
 
 func TestParsesStsEndpoint(t *testing.T) {
 	stsEndpoint := uuid.New().String()
+	command := parseTest(t, fmt.Sprintf("--sts-endpoint=%s", stsEndpoint))
+	require.EqualValues(t, stsEndpoint, command.STSEndpoint)
+}
+
+func TestParsesLogFilePath(t *testing.T) {
+	logFilePath := uuid.New().String()
+	command := parseTest(t, fmt.Sprintf("--log-file-path=%s", logFilePath))
+	require.EqualValues(t, logFilePath, command.LogFilePath)
+}
```

Comment:
As a result of boilerplate reduction/introduction of helper methods, basic parsing tests are much more concise.